api/webview/views.py

```diff
@@ -44,7 +45,25 @@ def perform_create(self, serializer):
     def get_queryset(self):
         """ Return queryset based on source
         """
-        return Document.objects.filter(source=self.kwargs['source'])
+        return Document.objects.filter(source=self.kwargs['source']).exclude(normalized=None)
+
+
+class DocumentsByProviderUpdatedDateTime(generics.ListAPIView):
+    """
+    List all documents updated within specified time frame
+    """
+    serializer_class = DocumentSerializer
+    permission_classes = (permissions.IsAuthenticatedOrReadOnly,)
+
+    def perform_create(self, serializer):
+        serializer.save(source=self.request.user)
+
+    def get_queryset(self):
+        """ Return queryset based on provider update time
+        """
+        queryset = Document.objects.all()
+        queryset = queryset.filter(providerUpdatedDateTime__gte=parse(self.kwargs['from'])).filter(providerUpdatedDateTime__lte=parse(self.kwargs['until']))
```

Comment:
This is awesome! Can you modify it so that instead of a new view, it checks for the argument in the normal document list view? It should add a filter only if the query parameter is there, and do nothing if it isn't.  Check out the source one that gloria added in her PR. 

Also, you can go ahead and add a filter on the initial queryset, instead of first querying for everything and then adding a filter (it will be faster!)

so, something like:

```
filters = {}
if self.kwargs.get('from', None):
    filters['from'] = self.kwargs['from']
   # other optional filters here

return queryset(**filters)
````